pkg/hpa/service.go

```diff
@@ -668,6 +668,10 @@ func (c *Service) UpdateHPAFromTortoiseRecommendation(ctx context.Context, torto
 		}
 
 		hpa = c.excludeExternalMetric(ctx, hpa)
+		if tortoiseName, ok := hpa.Annotations[annotation.TortoiseNameAnnotation]; ok {
```

Comment:
```suggestion
		if tortoiseName, ok := hpa.Annotations[annotation.TortoiseNameAnnotation]; ok {
		// If HPA has a deprecated annotation, we replace it with a new one. 
		// It allows us to remove a deprecated annotation completely later.
```

Better to add the context.